### PR TITLE
[RFC] Bypass poo#40958 zypper in curl failed in docker

### PR DESCRIPTION
**Pull request by michelmno**
if error detected on first trial, then report as soft_failure
and retry with --force-resolution.

Signed-off-by: Michel Normand <normand@linux.vnet.ibm.com>

- Related ticket: https://progress.opensuse.org/issues/40958
- Needles: none
- Verification run: in my local openQA PowerPC instance

### CODE REVIEW

File: tests/console/docker.pm [hunk 1]
```diff
@@ -100,8 +100,12 @@ sub run {
     die("error: missing container $container_name") unless ($output_containers =~ m/$container_name/);
 
     # containers state can be saved to a docker image
-    my $output = script_output("docker container exec $container_name zypper -n in curl", 300);
-    die('error: curl not installed in the container') unless ($output =~ m/Installing: curl.*done/);
+    my $output = script_run("docker container exec $container_name zypper -n in curl", 300);
```
Comment (SergioAtSUSE) [inline tests/console/docker.pm]: To avoid confusion here, please change the var name here to $exit_code

Comment (michelmno) [reply tests/console/docker.pm]: new commit updated to use exit_code var name as suggested

File: tests/console/docker.pm [hunk 2]
```diff
@@ -100,13 +100,19 @@ sub run {
...
+        record_soft_failure('poo#40958 - curl install failure, try with force-resolution.');
+        $exit_code = script_output("docker container exec $container_name zypper in --force-resolution -y -n curl", 300);
+        die('error: curl not installed in the container') unless ($exit_code =~ m/Installing: curl.*done/);
+    }
     assert_script_run("docker container commit $container_name tw:saved");
 
     # network is working inside of the containers
-    $output = script_output('docker container run tw:saved curl -I google.de');
-    die("network is not working inside of the container tw:saved") unless ($output =~ m{Location: http://www\.google\.de/});
+    $exit_code = script_output('docker container run tw:saved curl -I google.de');
```
Comment (SergioAtSUSE) [inline tests/console/docker.pm]: But, you also changed this var name, here it should be $output.

File: tests/console/docker.pm [hunk 3]
```diff
@@ -100,13 +100,19 @@ sub run {
...
+    die("network is not working inside of the container tw:saved") unless ($exit_code =~ m{Location: http://www\.google\.de/});
```
Comment (SergioAtSUSE) [inline tests/console/docker.pm]: And here too.

File: tests/console/docker.pm (hunk 2 above)
Comment (michelmno) [reply tests/console/docker.pm]: sorry for the iteration, commit with exit_code only where required.

Comment (michelmno) [reply tests/console/docker.pm]: resolved with last commit.

File: tests/console/docker.pm [hunk 4]
```diff
@@ -100,12 +100,18 @@ sub run {
     die("error: missing container $container_name") unless ($output_containers =~ m/$container_name/);
 
     # containers state can be saved to a docker image
-    my $output = script_output("docker container exec $container_name zypper -n in curl", 300);
-    die('error: curl not installed in the container') unless ($output =~ m/Installing: curl.*done/);
+    # note that here script_run do not return in case of failure,
+    # so returned value is more an exit_code than an output string.
```
Comment (okurz) [inline tests/console/docker.pm]: Not sure what you are referring to with your comment. The return value of `script_run` is not `more an exit_code` but is just that, an exit code :)

File: tests/console/docker.pm [hunk 5]
```diff
@@ -100,12 +100,18 @@ sub run {
...
+    my $exit_code = script_run("docker container exec $container_name zypper -n in curl", 300);
+    if ($exit_code) {
+        record_soft_failure('poo#40958 - curl install failure, try with force-resolution.');
```
Comment (okurz) [inline tests/console/docker.pm]: Please only reference product bugs with "soft failure". I think the workaround is a good idea though. How about you just replace `record_soft_failure` with `record_info`? Just like `record_info 'poo#40958', '…';`

File: tests/console/docker.pm (hunk 4 above)
Comment (michelmno) [reply tests/console/docker.pm]: comment removed.

File: tests/console/docker.pm (hunk 5 above)
Comment (michelmno) [reply tests/console/docker.pm]: use record_info as suggested

**Pull request merged**